### ISSUE TITLE
Exception when using extension

### ISSUE THREAD

**Issue by HDwares** [microsoft/vscode-java-test#1420]
Receiving BeanCreationException when running spring boot test using Test Runner for Java - VS Code extension. It was working fine till yesterday.

It is still working fine using eclipse' junit runner and also using maven test command.

Error description from debug console:
```
GenericWebApplicationContext: Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'springApplicationAdmjnRegistrar' defined in class path resource

SpringApplicationAdminJmxAutoConfiguration.class: Invocation of init method failed; nested exception is javax.management.InstanceAlreadyExistsException: org.springframework.boot:type=Admin,name=SpringApplication.

```

Please look into this issue, because this extension is very good only because of this issue I will again have to switch to other IDE for Java.

**Comment by jdneo** [comment]
Could you share a sample project which can repro this issue?

**Comment by HDwares** [comment]
Attached is a sample project. Execute the below test classes, you will receive same type of exception "Unable to create Application context, BeanCreationException:
1. src\test\java\com\example\SpringBootExamples\apiTests\ProductApiTestDummy.java
2. src\test\java\com\example\SpringBootExamples\unitTests\ProductControllerUnitTest.java

1st one was working fine earlier, 2nd one didn't work ever. After I tried the 2nd I started receiving error for both.

[SpringBootExamples.zip](https://github.com/microsoft/vscode-java-test/files/8718074/SpringBootExamples.zip)

**Comment by jdneo** [comment]
Thank you. I will take a look

**Comment by HDwares** [comment]
Fixed the 2nd test as well, now both tests run fine in eclipse, netbeans and intellij but are not working in vs code.

[SpringBootExamples.zip](https://github.com/microsoft/vscode-java-test/files/8730008/SpringBootExamples.zip)

**Comment by jdneo** [comment]
It's working on my side:

![image](https://user-images.githubusercontent.com/6193897/169432378-8648b4c7-d241-47f9-b65a-2201b8ee8778.png)

![image](https://user-images.githubusercontent.com/6193897/169432389-59713444-9637-428f-9594-0e4a0c3174a4.png)

Do you have any customized settings in your VS Code? Or any environment related settings?

**Comment by HDwares** [comment]
No, the only workspace vscode setting.json and launch.json which were autogenerated and which I provided in the zip. But I also tried removing them and also removed the user's setting.json, still receiving the same error:
<img width="872" alt="image" src="https://user-images.githubusercontent.com/49922820/169457902-4f84da71-25f6-4cfe-925a-b6356b3c6d53.png">

I in fact tried by using different machine as well so as to understand if there is some cache issue, but nope, same result. Strange thing is that 2nd test has started working now in vs code. Not able to understand where is the problem with first test.

**Comment by HDwares** [comment]
Attaching the full debug console log for better insight.
[debugConsole.txt](https://github.com/microsoft/vscode-java-test/files/8736254/debugConsole.txt)

**Comment by jdneo** [comment]
So far have no idea on this. I tried several times but could not repro the issue. 🤔

**Comment by HDwares** [comment]
If you could tell where can I see the arguments with which the test is running, it would be very helpful. Because I found that on disabling the startSpringBootApplication line and starting application separately test runs fine.

**Comment by HDwares** [comment]
Latest update: when I disabled the actuator dependency in pom.xml and also disabled the actuator related classes, all the tests passed.

So, the extension is generating some config when actuator is enabled, due to which it tried to create it's own context before starting the application thus the exception.

**Comment by jdneo** [comment]
> So, the extension is generating some config when actuator is enabled, due to which it tried to create it's own context before starting the application thus the exception.

I don't think so. I didn't disable that dependency and the test passes. At least the test runner extension won't do anything related with actuator.

**Comment by HDwares** [comment]
Can you try after installing spring boot extension pack?

ext install Pivotal.vscode-boot-dev-pack

**Comment by jdneo** [comment]
Thank you @HDwares for the update.

BTW, do you think that's a bug of the Spring Boot Tools extension, or it's as expected??

**Comment by HDwares** [comment]
I think this is a feature and not a bug though I didn't find any documentation for that setting on their extension page. Also I don't know what exactly the feature is, how 'true' value in that setting will help a user.
